ext/handlers/callbackquery.go

```diff
@@ -11,27 +12,31 @@ import (
 type CallbackQuery struct {
 	AllowChannel bool
 	Filter       filters.CallbackQuery
-	Response     Response
+	Response     func(b *gotgbot.Bot, ctx *ext.Context) error
 }
 
-func NewCallback(filter filters.CallbackQuery, r Response) CallbackQuery {
-	return CallbackQuery{
+func NewCallback(filter filters.CallbackQuery, response func(b *gotgbot.Bot, ctx *ext.Context) error) *CallbackQuery {
+	return &CallbackQuery{
 		Filter:   filter,
-		Response: r,
+		Response: response,
 	}
 }
 
-// SetAllowChannel Enables channel messages for this handler.
-func (cb CallbackQuery) SetAllowChannel(allow bool) CallbackQuery {
+// SetAllowChannel enables channel messages for this handler.
+func (cb *CallbackQuery) SetAllowChannel(allow bool) *CallbackQuery {
```

Comment:
Why are we changing to a pointer reference here?

Reply:
Changing the receiver from CallbackQuery to *CallbackQuery in the SetAllowChannel method allows the method to modify the CallbackQuery struct directly, rather than working with a copy of the struct.

Reply:
I understand the function of the change, but this doesnt explain the why.

Currently, SetAllowChannel copies the receiver and returns it, allowing for immutable chaining of methods. Changing it to be a pointer would break the immutability and introduce side effects to the method, which would then be inconsistent with all other handlers.